spec/models/article_spec.rb

```diff
@@ -528,6 +528,18 @@ def build_and_validate_article(*args)
       article.organization = build(:organization)
       expect(article.username).to eq(article.organization.slug)
     end
+
+    it "returns a downcased username when user has uppercase characters" do
+      upcased_user = create(:user, username: "UpcasedUserName")
+      article = create(:article, user: upcased_user)
+      expect(article.username).not_to match(/[AZ]+/)
+    end
+
+    it "returns a downcased username when an org slug has uppercase characters" do
+      upcased_org = create(:organization, slug: "UpcasedUserName")
+      article = create(:article, organization: upcased_org)
+      expect(article.username).not_to match(/[AZ]+/)
+    end
```

Comment:
These tests make sure the `before_save` callback will "sanitize" the path regardless of the capitalization of the User/Org